lib/rack/test.rb

```diff
@@ -48,96 +55,58 @@ class Session
-      # if given.
       #
-      # Example:
-      #   get "/"
-      def get(uri, params = {}, env = {}, &block)
-        custom_request('GET', uri, params, env, &block)
-      end
-
-      # Issue a POST request for the given URI. See #get
+      # The following methods are defined via metaprogramming: get, post, put, patch,
+      # delete, options, and head. Each method submits a request with the given request
+      # method, with the given URI and optional parameters and rack environment.
+      # Examples:
       #
-      # Example:
-      #   post "/signup", "name" => "Bryan"
-      def post(uri, params = {}, env = {}, &block)
-        custom_request('POST', uri, params, env, &block)
-      end
-
-      # Issue a PUT request for the given URI. See #get
+      #   # URI only:
+      #   get("/")                   # GET /
+      #   get("/?foo=bar")           # GET /?foo=bar
       #
-      # Example:
-      #   put "/"
-      def put(uri, params = {}, env = {}, &block)
-        custom_request('PUT', uri, params, env, &block)
-      end
-
-      # Issue a PATCH request for the given URI. See #get
+      #   # URI and parameters
+      #   get("/foo", 'bar'=>'baz')  # GET /foo?bar=baz
+      #   post("/foo", 'bar'=>'baz') # POST /foo (bar=baz in request body)
       #
-      # Example:
-      #   patch "/"
-      def patch(uri, params = {}, env = {}, &block)
-        custom_request('PATCH', uri, params, env, &block)
-      end
-
-      # Issue a DELETE request for the given URI. See #get
+      #   # URI, parameters, and rack environment
+      #   get("/bar", {}, 'CONTENT_TYPE'=>'foo')
+      #   get("/bar", {'foo'=>'baz'}, 'HTTP_ACCEPT'=>'*')
       #
-      # Example:
-      #   delete "/"
-      def delete(uri, params = {}, env = {}, &block)
-        custom_request('DELETE', uri, params, env, &block)
-      end
+      # The above methods as well as #request and #custom_request store the Rack::Request
+      # submitted in #last_request. The methods store a Rack::MockResponse based on the
+      # response in #last_response. #last_response is also returned by the methods.
+      # If a block is given, #last_response is also yielded to the block.
+      def initialize(mock_session)
+        mock_session = MockSession.new(mock_session) unless mock_session.is_a?(MockSession)
 
-      # Issue an OPTIONS request for the given URI. See #get
-      #
-      # Example:
-      #   options "/"
-      def options(uri, params = {}, env = {}, &block)
-        custom_request('OPTIONS', uri, params, env, &block)
+        @env = {}
+        @digest_username = nil
+        @digest_password = nil
+        @rack_mock_session = mock_session
       end
 
-      # Issue a HEAD request for the given URI. See #get
-      #
-      # Example:
-      #   head "/"
-      def head(uri, params = {}, env = {}, &block)
-        custom_request('HEAD', uri, params, env, &block)
+      %w[get post put patch delete options head].each do |meth|
```

Comment:
I prefer to use `method` or `method_name` (better).

Reply:
OK, I can make that change.